### DISCUSSION TITLE
RoleName translation

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6819] [General]
Hi,

in my serene test application, i've inserted three roles on database

1 - System Administrator
2 - Tenant Administrator
3 - Customer

I would these three items translated in other language, as Italian.

What is the best practices to do it in serene, in your opinion?

Thank you.

**Comment by VictorTomaili** [comment]
Check the [category](https://demo.serenity.is/Northwind/Category#) sample for localization. Open a dialog and press the localization button.

**Reply by elenaaralla** [reply]
Hi @VictorTomaili 

I tryed to translate Category "Beverage" on online demo, but I am not able to see the translation (when I try to insert a product, for example)

I chose Italian as the language of the application.

see attached short video.

https://github.com/serenity-is/Serenity/assets/14870650/eab989fa-4313-4122-879d-aa6a42b90ace

**Reply by VictorTomaili** [reply]
These translations use a translation table. Like `ProductLang`. To show them in serene application, you need to include `Localizations` in the service request. You may need to create a custom lookup which retrieves these translations from the `ProductLang` table. Check the `LocalizationBehavior` to understand how you can do this. To show it in a grid, you can implement a formatter which retrieves the value of a specific language. To show it in lookup, you can override `itemText` for your customized lookup editor.
